src/finn/transformation/fpgadataflow/convert_to_hls_layers.py

```diff
@@ -128,105 +132,130 @@ def apply(self, model):
                     )
                     graph.node.insert(node_ind, padding_node)
 
-                # Ensure that only supported HLS nodes are inserted
-                is_square_image = ConvInpGen_idim_h == ConvInpGen_idim_w
-                is_square_kernel = k_h == k_w
-                is_kernel_pointwise = k_h == 1 and k_w == 1
-                is_equal_stride = stride_h == stride_w
-                is_1d_convolution = (k_h == 1 and k_w > 1 and ifm_dim_h == 1) or (
-                    k_h > 1 and k_w == 1 and ifm_dim_w == 1
-                )
-
-                if (stride_h > 1 or stride_w > 1) and is_kernel_pointwise:
-                    assert is_square_image, (
-                        "%s : DownSampler currently only supports square input images."
-                        % n.name
-                    )
-                    assert is_equal_stride, (
-                        """%s : DownSampler currently only supports equal stride value
-                        along different axes."""
-                        % n.name
-                    )
-                    ConvInpGen_idim = ConvInpGen_idim_h
-                    stride = stride_h
-                    # create DownSampler node
+                if self.use_rtl_variant:
```

Comment:
e.g. `test_convert_to_hls_conv_layer` seems to suggest 1x1 kernels are not supported?
```
    if use_rtl_swg and kernel_size == 1:
        pytest.skip("1x1 kernel not supported by current RTL SWG")

```

Reply:
I added a check for the 1x1 kernel case (which will be supported later using "parallel mode") so that the conversion falls back to HLS with a warning.

Other existing checks or skip conditions target generally-illegal convolution hyperparameters or unsupported folding factors, both of which are not really appropriate to check during HLS conversion.